Library/Homebrew/readall.rb

```diff
@@ -0,0 +1,79 @@
+require "formula"
+require "tap"
+require "thread"
+require "readall"
+
+module Readall
+  class << self
+    def valid_ruby_syntax?(ruby_files)
+      failed = false
+      workers = (0...Hardware::CPU.cores).map do
+        Thread.new do
+          begin
+            while rb = ruby_files.pop(true)
+              # As a side effect, print syntax errors/warnings to `$stderr`.
+              failed = true if syntax_errors_or_warnings?(rb)
+            end
+          rescue ThreadError # ignore empty queue error
+          end
+        end
+      end
+      workers.each(&:join)
+      !failed
+    end
+
+    def valid_aliases?(alias_dirs)
+      failed = false
+      alias_dirs.each do |alias_dir|
+        next unless alias_dir.directory?
+        Pathname.glob("#{alias_dir}/*").each do |f|
```

Comment:
You're right. I'm leaning towards making relatively safe changes during the rearrangement, but I can see the argument for just shuffling things around and keeping the actual changes to the bare minimum.